Dockerfile

```diff
@@ -1,5 +1,5 @@
 # Build the manager binary
-FROM docker.io/golang:1.17 as builder
+FROM docker.io/golang:1.18 as builder
```

Comment:
LGTM. 
However, when building samba-operator and running `make check` with `go1.18` there are few errors. I will submit a separate PR with fixes on-top of this one.